new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java

```diff
@@ -14,30 +21,225 @@
+     */
+    public static boolean deleteOperator() { return Operator.delete(); }
+
     // ----------------------   domain  -----------------------------------
 
+    /**
+     * Create domain custom resource from the given domain yaml file.
+     * @param domainUID
+     * @param namespace
+     * @param domainYAML
+     * @return true on success, false otherwise
+     */
     public static boolean createDomainCustomResource(String domainUID, String namespace, String domainYAML) {
         return Domain.createDomainCustomResource(domainUID, namespace, domainYAML);
     }
 
+    /**
+     * Shutdown the domain
+     * @param domainUID
+     * @param namespace
+     * @return true on success, false otherwise
+     */
+    public static boolean shutdown(String domainUID, String namespace) {
+        return Domain.shutdown(domainUID, namespace);
+    }
+
+    /**
+     *
+     * @param domainUID
+     * @param namespace
+     * @return true on success, false otherwise
+     */
+    public static boolean restart(String domainUID, String namespace) {
+        return Domain.restart(domainUID, namespace);
+    }
+
+    /**
+     *
+     * @param domainUID
+     * @param namespace
+     * @return true on success, false otherwise
+     */
+    public static boolean deleteDomainCustomResource(String domainUID, String namespace) {
+        return Domain.deleteDomainCustomResource(domainUID, namespace);
+    }
+
     // ------------------------   ingress controller ----------------------
 
+    /**
+     * Install Traefik Operator
+     * @param valuesYaml values yaml file to be used
+     * @return true on success, false otherwise
+     */
+    public static boolean installTraefik(String valuesYaml) {
+        return Traefik.install(valuesYaml);
+    }
 
+    /**
+     * Create Treafik Ingress
+     * @param valuesYaml values yaml file to be used
+     * @return true on success, false otherwise
+     */
+    public static boolean createIngress(String valuesYaml) {
+        return Traefik.createIngress(valuesYaml);
+    }
 
     // -------------------------  namespaces -------------------------------
 
+    /**
+     * Create Kubernetes namespace
+     * @param name the name of the namespace
+     * @return true on success, false otherwise
+     */
     public static boolean createNamespace(String name) {
         return Namespace.createNamespace(name);
     }
 
+    /**
+     * Create a namespace with unique name
+     * @return true on success, false otherwise
+     */
     public static String createUniqueNamespace() {
         return Namespace.createUniqueNamespace();
     }
 
+    // -------------------------   pv/pvc  ---------------------------------
+
+    /**
+     * Create Kubernetes Persistent Volume using the yaml provided
+     * @param pvYaml the persistent volume yaml file
+     * @return true on success, false otherwise
+     */
+    public static boolean createPersistentVolume(String pvYaml) {
```

Comment:
We have an API for standard types. `io.kubernetes.client.openapi.models.V1PersistentVolume`

Reply:
happy to discuss, but if it is ugly, i would prefer to have a fluent api that hides that ugliness under the covers

Reply:
These types already provide a fluent API: https://github.com/kubernetes-client/java/blob/master/examples/src/main/java/io/kubernetes/client/examples/FluentExample.java

Reply:
prefect :)

Reply:
Shall I remove the action methods for all standard types?